app/src/main/java/de/danoeh/antennapod/activity/SingleFileAddActivity.java

```diff
@@ -0,0 +1,220 @@
+import android.content.pm.PackageManager;
+import android.media.MediaMetadataRetriever;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Environment;
+import android.support.annotation.Nullable;
+import android.support.v4.app.ActivityCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.afollestad.materialdialogs.MaterialDialog;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.feed.FeedItem;
+import de.danoeh.antennapod.core.feed.FeedMedia;
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+import de.danoeh.antennapod.core.storage.DBTasks;
+import de.danoeh.antennapod.core.util.IntentUtils;
+import de.danoeh.antennapod.core.util.StorageUtils;
+
+/**
+ * Lets the user start the single file import process from a path
+ */
+public class SingleFileAddActivity extends AppCompatActivity {
+
+    private static final String TAG = "SingleFileAddAct";
+    private static final int PERMISSION_REQUEST_READ_EXTERNAL_STORAGE = 5;
+
+    private static final int CHOOSE_SINGLE_FILE = 1;
+
+    private Intent intentPickAction;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        setTheme(UserPreferences.getTheme());
+        super.onCreate(savedInstanceState);
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        setContentView(R.layout.singlefile_add);
+
+        final TextView singleFileHeaderExplanation1 = findViewById(R.id.singleFileHeadingExplanation1);
+        final TextView singleFileExplanation1 = findViewById(R.id.singleFileExplanation1);
+
+        Button butChooseFilesystem = findViewById(R.id.butChooseFileFromFilesystem);
+        butChooseFilesystem.setOnClickListener(v -> chooseFileFromFilesystem());
+
+        int nextOption = 1;
+        String optionLabel = getString(R.string.singlefile_import_option);
+        intentPickAction = new Intent(Intent.ACTION_PICK);
+
+        if(!IntentUtils.isCallable(getApplicationContext(), intentPickAction)) {
+            intentPickAction.setData(null);
+            if(!IntentUtils.isCallable(getApplicationContext(), intentPickAction)) {
+                singleFileHeaderExplanation1.setVisibility(View.GONE);
+                singleFileExplanation1.setVisibility(View.GONE);
+                findViewById(R.id.divider1).setVisibility(View.GONE);
+                butChooseFilesystem.setVisibility(View.GONE);
+            }
+        }
+        if(singleFileExplanation1.getVisibility() == View.VISIBLE) {
+            singleFileHeaderExplanation1.setText(String.format(optionLabel, nextOption));
+            nextOption++;
```

Comment:
What is this `nextOption` variable for?

Reply:
Also copied from `OpmlImportFromPathActivity`. There are several sections in the layout and in `onCreate()` increments `nextOption` to set the "explanation header".

At first I thought there might be other sections, but right now I'm not really sure what those would be (when we're able to import a whole folder with the same button, for example).